db/compaction/compaction_picker_fifo.cc

```diff
@@ -79,10 +79,9 @@ Compaction* FIFOCompactionPicker::PickTTLCompaction(
       FileMetaData* f = *ritr;
       assert(f);
       if (f->fd.table_reader && f->fd.table_reader->GetTableProperties()) {
-        uint64_t creation_time =
-            f->fd.table_reader->GetTableProperties()->creation_time;
-        if (creation_time == 0 ||
-            creation_time >= (current_time - mutable_cf_options.ttl)) {
+        uint64_t est_newest_key_time = f->TryGetNewestKeyTime();
+        if (est_newest_key_time == kUnknownNewestKeyTime ||
```

Comment:
You are right. I pushed another commit to fix this. I originally was using the `creation_time` fallback, and then I made the refactor to use `TryGetNewestTime`, but `TryGetNewestTime` does not handle the fallback to `creation_time` in this case because we are not using the `prev_file`'s `creation_time`